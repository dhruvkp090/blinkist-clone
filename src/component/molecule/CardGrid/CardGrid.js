import React from 'react';
import BookCard from '../Card/BookCard';
import Grid from '@material-ui/core/Grid';

export const CardGrid = (props) => {
    
    return (
        <Grid container spacing={3}>
            {props.bookList.map((book, index) => (
                <Grid key={index} item xs={12} sm={6} md={4}>
                    <BookCard 
                        id={book.id} 
                        bookTitle={book.bookTitle} 
                        bookAuthor={book.bookAuthor} 
                        bookDuration={book.bookDuration} 
                        isFinished={book.isFinished} 
                        img={book.img} 
                        isInLibrary = {props.isInLibrary}
                        addToLibrary={props.addToLibrary}
                        onClick={(e) => props.onClick(e, book.id)}
                    />
                </Grid>
            ))}
        </Grid>
    );
};

export default CardGrid;
