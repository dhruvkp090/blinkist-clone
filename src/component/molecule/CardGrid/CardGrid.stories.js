import React from 'react';
import CardGrid from './CardGrid';

export default {
    title: "Molecule/CardGrid",
    component: CardGrid,
};

const Template = args => (<CardGrid {...args} />);

const test = () => {
    return Math.random() < 0.5;
};

export const Default = Template.bind({});
Default.args = {
    bookList: [
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "1984",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
    ],
    isInLibrary: test
};
