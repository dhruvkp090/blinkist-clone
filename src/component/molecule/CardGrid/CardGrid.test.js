import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react';
import CardGrid from './CardGrid';
import '@testing-library/jest-dom';

const isInLibraryMock = jest.fn();
const onClickMock = jest.fn();
const  bookList= [
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "Animal Farm",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
        {
            bookTitle: "1984",
            bookAuthor: "Gorge Orwell",
            bookDuration: "20",
            isFinished: false
        },
    ];

it('Snapshot with default image and finished book', () => {
    const { asFragment } = render(<CardGrid isInLibrary={isInLibraryMock.mockReturnValue(true)} bookList={bookList} />);

    expect(asFragment()).toMatchSnapshot();
});

it('Click event works', () => {
    const { rerender } = render(<CardGrid isInLibrary={isInLibraryMock.mockReturnValue(true)} isFinished={false} bookList={bookList} onClick={onClickMock} />);
    screen.getAllByText("Mark as Finished");
    fireEvent.click(screen.getAllByText('Mark as Finished')[0]);
});