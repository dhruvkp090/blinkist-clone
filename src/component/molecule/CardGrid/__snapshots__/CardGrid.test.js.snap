// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot with default image and finished book 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <img
          alt="Animal Farm"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
          src="https://images.blinkist.com/images/books/534d4cb46633610007320000/1_1/640.jpg"
          title="Animal Farm"
        />
        <div
          class="MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Animal Farm
          </h6>
          <p
            class="MuiTypography-root author MuiTypography-body1 MuiTypography-colorSecondary MuiTypography-gutterBottom"
          >
            <strong>
              Gorge Orwell
            </strong>
          </p>
          <div
            class="timeSection"
          >
            <img
              alt=""
              class="icon"
              src="https://d17pjsg7x52x9r.cloudfront.net/assets/components/book_card/clock-b0e2e0235fbe1df824d662b2b3b96611e3711bf5b5c7556b8bd3828720f86dbc.svg"
            />
            <p
              class="MuiTypography-root timeText MuiTypography-body2 MuiTypography-gutterBottom"
            >
              20-minute read
            </p>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            role="button"
            style="justify-content: center;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                <strong>
                  Mark as Finished
                </strong>
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <img
          alt="Animal Farm"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
          src="https://images.blinkist.com/images/books/534d4cb46633610007320000/1_1/640.jpg"
          title="Animal Farm"
        />
        <div
          class="MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Animal Farm
          </h6>
          <p
            class="MuiTypography-root author MuiTypography-body1 MuiTypography-colorSecondary MuiTypography-gutterBottom"
          >
            <strong>
              Gorge Orwell
            </strong>
          </p>
          <div
            class="timeSection"
          >
            <img
              alt=""
              class="icon"
              src="https://d17pjsg7x52x9r.cloudfront.net/assets/components/book_card/clock-b0e2e0235fbe1df824d662b2b3b96611e3711bf5b5c7556b8bd3828720f86dbc.svg"
            />
            <p
              class="MuiTypography-root timeText MuiTypography-body2 MuiTypography-gutterBottom"
            >
              20-minute read
            </p>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            role="button"
            style="justify-content: center;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                <strong>
                  Mark as Finished
                </strong>
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <img
          alt="Animal Farm"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
          src="https://images.blinkist.com/images/books/534d4cb46633610007320000/1_1/640.jpg"
          title="Animal Farm"
        />
        <div
          class="MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Animal Farm
          </h6>
          <p
            class="MuiTypography-root author MuiTypography-body1 MuiTypography-colorSecondary MuiTypography-gutterBottom"
          >
            <strong>
              Gorge Orwell
            </strong>
          </p>
          <div
            class="timeSection"
          >
            <img
              alt=""
              class="icon"
              src="https://d17pjsg7x52x9r.cloudfront.net/assets/components/book_card/clock-b0e2e0235fbe1df824d662b2b3b96611e3711bf5b5c7556b8bd3828720f86dbc.svg"
            />
            <p
              class="MuiTypography-root timeText MuiTypography-body2 MuiTypography-gutterBottom"
            >
              20-minute read
            </p>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            role="button"
            style="justify-content: center;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                <strong>
                  Mark as Finished
                </strong>
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <img
          alt="Animal Farm"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
          src="https://images.blinkist.com/images/books/534d4cb46633610007320000/1_1/640.jpg"
          title="Animal Farm"
        />
        <div
          class="MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Animal Farm
          </h6>
          <p
            class="MuiTypography-root author MuiTypography-body1 MuiTypography-colorSecondary MuiTypography-gutterBottom"
          >
            <strong>
              Gorge Orwell
            </strong>
          </p>
          <div
            class="timeSection"
          >
            <img
              alt=""
              class="icon"
              src="https://d17pjsg7x52x9r.cloudfront.net/assets/components/book_card/clock-b0e2e0235fbe1df824d662b2b3b96611e3711bf5b5c7556b8bd3828720f86dbc.svg"
            />
            <p
              class="MuiTypography-root timeText MuiTypography-body2 MuiTypography-gutterBottom"
            >
              20-minute read
            </p>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            role="button"
            style="justify-content: center;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                <strong>
                  Mark as Finished
                </strong>
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <img
          alt="Animal Farm"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
          src="https://images.blinkist.com/images/books/534d4cb46633610007320000/1_1/640.jpg"
          title="Animal Farm"
        />
        <div
          class="MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Animal Farm
          </h6>
          <p
            class="MuiTypography-root author MuiTypography-body1 MuiTypography-colorSecondary MuiTypography-gutterBottom"
          >
            <strong>
              Gorge Orwell
            </strong>
          </p>
          <div
            class="timeSection"
          >
            <img
              alt=""
              class="icon"
              src="https://d17pjsg7x52x9r.cloudfront.net/assets/components/book_card/clock-b0e2e0235fbe1df824d662b2b3b96611e3711bf5b5c7556b8bd3828720f86dbc.svg"
            />
            <p
              class="MuiTypography-root timeText MuiTypography-body2 MuiTypography-gutterBottom"
            >
              20-minute read
            </p>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            role="button"
            style="justify-content: center;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                <strong>
                  Mark as Finished
                </strong>
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <img
          alt="Animal Farm"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
          src="https://images.blinkist.com/images/books/534d4cb46633610007320000/1_1/640.jpg"
          title="Animal Farm"
        />
        <div
          class="MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Animal Farm
          </h6>
          <p
            class="MuiTypography-root author MuiTypography-body1 MuiTypography-colorSecondary MuiTypography-gutterBottom"
          >
            <strong>
              Gorge Orwell
            </strong>
          </p>
          <div
            class="timeSection"
          >
            <img
              alt=""
              class="icon"
              src="https://d17pjsg7x52x9r.cloudfront.net/assets/components/book_card/clock-b0e2e0235fbe1df824d662b2b3b96611e3711bf5b5c7556b8bd3828720f86dbc.svg"
            />
            <p
              class="MuiTypography-root timeText MuiTypography-body2 MuiTypography-gutterBottom"
            >
              20-minute read
            </p>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            role="button"
            style="justify-content: center;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                <strong>
                  Mark as Finished
                </strong>
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <img
          alt="Animal Farm"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
          src="https://images.blinkist.com/images/books/534d4cb46633610007320000/1_1/640.jpg"
          title="Animal Farm"
        />
        <div
          class="MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Animal Farm
          </h6>
          <p
            class="MuiTypography-root author MuiTypography-body1 MuiTypography-colorSecondary MuiTypography-gutterBottom"
          >
            <strong>
              Gorge Orwell
            </strong>
          </p>
          <div
            class="timeSection"
          >
            <img
              alt=""
              class="icon"
              src="https://d17pjsg7x52x9r.cloudfront.net/assets/components/book_card/clock-b0e2e0235fbe1df824d662b2b3b96611e3711bf5b5c7556b8bd3828720f86dbc.svg"
            />
            <p
              class="MuiTypography-root timeText MuiTypography-body2 MuiTypography-gutterBottom"
            >
              20-minute read
            </p>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            role="button"
            style="justify-content: center;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                <strong>
                  Mark as Finished
                </strong>
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <img
          alt="Animal Farm"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
          src="https://images.blinkist.com/images/books/534d4cb46633610007320000/1_1/640.jpg"
          title="Animal Farm"
        />
        <div
          class="MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Animal Farm
          </h6>
          <p
            class="MuiTypography-root author MuiTypography-body1 MuiTypography-colorSecondary MuiTypography-gutterBottom"
          >
            <strong>
              Gorge Orwell
            </strong>
          </p>
          <div
            class="timeSection"
          >
            <img
              alt=""
              class="icon"
              src="https://d17pjsg7x52x9r.cloudfront.net/assets/components/book_card/clock-b0e2e0235fbe1df824d662b2b3b96611e3711bf5b5c7556b8bd3828720f86dbc.svg"
            />
            <p
              class="MuiTypography-root timeText MuiTypography-body2 MuiTypography-gutterBottom"
            >
              20-minute read
            </p>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            role="button"
            style="justify-content: center;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                <strong>
                  Mark as Finished
                </strong>
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <img
          alt="Animal Farm"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
          src="https://images.blinkist.com/images/books/534d4cb46633610007320000/1_1/640.jpg"
          title="Animal Farm"
        />
        <div
          class="MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Animal Farm
          </h6>
          <p
            class="MuiTypography-root author MuiTypography-body1 MuiTypography-colorSecondary MuiTypography-gutterBottom"
          >
            <strong>
              Gorge Orwell
            </strong>
          </p>
          <div
            class="timeSection"
          >
            <img
              alt=""
              class="icon"
              src="https://d17pjsg7x52x9r.cloudfront.net/assets/components/book_card/clock-b0e2e0235fbe1df824d662b2b3b96611e3711bf5b5c7556b8bd3828720f86dbc.svg"
            />
            <p
              class="MuiTypography-root timeText MuiTypography-body2 MuiTypography-gutterBottom"
            >
              20-minute read
            </p>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            role="button"
            style="justify-content: center;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                <strong>
                  Mark as Finished
                </strong>
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <img
          alt="Animal Farm"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
          src="https://images.blinkist.com/images/books/534d4cb46633610007320000/1_1/640.jpg"
          title="Animal Farm"
        />
        <div
          class="MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Animal Farm
          </h6>
          <p
            class="MuiTypography-root author MuiTypography-body1 MuiTypography-colorSecondary MuiTypography-gutterBottom"
          >
            <strong>
              Gorge Orwell
            </strong>
          </p>
          <div
            class="timeSection"
          >
            <img
              alt=""
              class="icon"
              src="https://d17pjsg7x52x9r.cloudfront.net/assets/components/book_card/clock-b0e2e0235fbe1df824d662b2b3b96611e3711bf5b5c7556b8bd3828720f86dbc.svg"
            />
            <p
              class="MuiTypography-root timeText MuiTypography-body2 MuiTypography-gutterBottom"
            >
              20-minute read
            </p>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            role="button"
            style="justify-content: center;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                <strong>
                  Mark as Finished
                </strong>
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <img
          alt="1984"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
          src="https://images.blinkist.com/images/books/534d4cb46633610007320000/1_1/640.jpg"
          title="1984"
        />
        <div
          class="MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
          >
            1984
          </h6>
          <p
            class="MuiTypography-root author MuiTypography-body1 MuiTypography-colorSecondary MuiTypography-gutterBottom"
          >
            <strong>
              Gorge Orwell
            </strong>
          </p>
          <div
            class="timeSection"
          >
            <img
              alt=""
              class="icon"
              src="https://d17pjsg7x52x9r.cloudfront.net/assets/components/book_card/clock-b0e2e0235fbe1df824d662b2b3b96611e3711bf5b5c7556b8bd3828720f86dbc.svg"
            />
            <p
              class="MuiTypography-root timeText MuiTypography-body2 MuiTypography-gutterBottom"
            >
              20-minute read
            </p>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            role="button"
            style="justify-content: center;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                <strong>
                  Mark as Finished
                </strong>
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
